tasks/main.yml

```diff
@@ -86,14 +86,26 @@
   template: src='pxe_nodes.json.j2' dest='/var/www/provision/nodes/pxe_nodes.json' backup=yes
   tags: pxe_data
 
-- name: create_kickstart_group_files
+- name: create_kickstart_group_files for legacy nodes
   template: src=kickstart.j2 dest="/var/www/html/kickstart/{{item}}.ks" validate='/usr/bin/ksvalidator %s' backup=yes
   tags: dhcp_kickstart_config
   with_items: "{{ groups|sort }}"
   when:
     - item not in dhcp_kickstart_skip_these_groups
     - groups[item][0] is defined
     - hostvars[groups[item][0]]['os_disks'] is defined
+    - hostvars[groups[item][0]]['target_os'] is undefined
```

Comment:
`hostvars[groups[item][0]]['target_os'] is undefined or hostvars[groups[item][0]]['target_os'] == "centos_7"`

Reply:
Oikeastaan paras voisi olla, että siellä on yksi taski "is undefined" ja sitten erikseen jokaista pyydettävissä olevaa versiota varten oma taski jossa katsotaan mikä stringi muuttujassa on. Muutos tämänhetkiseen olisi siis erillisen taskin lisääminen centos_7 kickstartin luomiseen ja sillä olisi alustavasti sama sisältö kuin tuolla "is undefined" -taskilla.

Reply:
Melkein siellä voisi olla sellainenkin, että jos "is defined", mutta arvo ei ole mikään meidän tuntemista, niin keskeytetään ja annetaan virheilmoitus - nykyisellään silloin kickstart-file jää luomatta.

Reply:
Nyt on main-tiedoston alussa tarkistus, että target_os ei ole muuta kuin centos7 tai centos_8stream. Jos on muuta, niin tulee hylätty. Ja target_os on centos7 tai centos8_stream, rooli tekee sen mukaisen templaten perusteella kickstart-tiedostot.